mart/attack/adversary.py

```diff
@@ -159,7 +159,13 @@ def on_run_start(
         self.perturber(input, target)
 
         # Split param groups by input elements, so that we can schedule optimizers individually.
-        param_groups = [{"params": [param]} for param in self.perturber.parameters()]
+        if hasattr(self.perturber, "parameters_optim"):
+            # param_groups with learning rate and other optim params.
+            param_groups = self.perturber.parameters_optim()
+        else:
+            # Backward compatibility.
+            param_groups = [{"params": [param]} for param in self.perturber.parameters()]
```

Comment:
Why is this necessary of all perturbers have changed to the new interface?

Reply:
Deleted.